### ISSUE TITLE
Speech To Text Batch Transcription Support for more formats

### ISSUE THREAD

**Issue by chassq** [Azure-Samples/cognitive-services-speech-sdk#2114]
Currently it looks like the Batch transcribe feature supports only mp3, ogg and wav formats. The issue is we record calls in MS Teams and the format is mp4. Could you please add the functionality to transcribe from mp4? If not, are there any alternatives. We do not want to leave the azure cloud to perform transcoding if possible as the data could be of a sensitive nature.

**Comment by chschrae** [comment]
Hi there,

mp4 is a multimedia format which may or may not contain audio. It won't be directly supported by the SDK anytime soon. The only alternative would be to have some sort of converter running in an azure function or something like that. You could then write the output to some azure blob storage.

There is a way to turn on transcription for the meetings that stores it with the video though:
https://support.microsoft.com/en-us/office/view-edit-and-manage-video-transcripts-and-captions-3cb9acb6-05b2-4f59-a50d-7df61123aa20#bkmk_how-captions-and-transcripts-are-stored
https://learn.microsoft.com/en-us/microsoftteams/meeting-transcription-captions

Does that solve your problem?

**Comment by chassq** [comment]
Unfortunately the teams transcripts are having accuracy issuez for us so we wanted to pass in the actual audio to check if the batch transcripts faired better. We would not want to pass the data out of the azure cloud due to security reasons. Any good nuget pkgs?

**Comment by chschrae** [comment]
Yes, there are several NuGet packages that can extract audio from an MP4 file. Here are a few options:

1. **[FFMediaToolkit 4.5.1](https://www.nuget.org/packages/FFMediaToolkit/)**¹: This is a cross-platform audio/video processing library based on FFmpeg native libraries. It supports audio/video frames extraction, reading file metadata, and encoding media files from bitmap images and audio data¹.

2. **FFMpegCore 5.[1](https://www.nuget.org/packages/FFMediaToolkit/).0**²: This is a .NET Standard FFMpeg/FFProbe wrapper for easily integrating media analysis and conversion into your .NET applications. It supports both synchronous and asynchronous calls API². You can use the `FFMpeg.ExtractAudio` method to extract the audio track from a video file².

Please note that you should check the compatibility of these packages with your project's .NET version before using them.
